packages/contracts-periphery/contracts/L2/messaging/L2ERC721Bridge.sol

```diff
@@ -0,0 +1,196 @@
+        // slither-disable-next-line reentrancy-events
+        IL2StandardERC721(_l2Token).burn(msg.sender, _tokenId);
+
+        // Construct calldata for l1ERC721Bridge.finalizeERC721Withdrawal(_to, _tokenId)
+        // slither-disable-next-line reentrancy-events
+        address l1Token = IL2StandardERC721(_l2Token).l1Token();
+        bytes memory message = abi.encodeWithSelector(
+            IL1ERC721Bridge.finalizeERC721Withdrawal.selector,
+            l1Token,
+            _l2Token,
+            _from,
+            _to,
+            _tokenId,
+            _data
+        );
+
+        // Send message to L1 bridge
+        // slither-disable-next-line reentrancy-events
+        sendCrossDomainMessage(l1ERC721Bridge, _l1Gas, message);
+
+        // slither-disable-next-line reentrancy-events
+        emit WithdrawalInitiated(l1Token, _l2Token, msg.sender, _to, _tokenId, _data);
+    }
+
+    /************************************
+     * Cross-chain Function: Depositing *
+     ************************************/
+
+    /**
+     * @inheritdoc IL2ERC721Bridge
+     */
+    function finalizeDeposit(
+        address _l1Token,
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _tokenId,
+        bytes calldata _data
+    ) external virtual onlyFromCrossDomainAccount(l1ERC721Bridge) {
+        // Check the target token is compliant and
+        // verify the deposited token on L1 matches the L2 deposited token representation here
+        if (
+            // slither-disable-next-line reentrancy-events
+            ERC165Checker.supportsInterface(_l2Token, 0x1d1d8b63) &&
+            _l1Token == IL2StandardERC721(_l2Token).l1Token()
+        ) {
+            // When a deposit is finalized, we give the NFT with the same tokenId to the account
+            // on L2.
+            // slither-disable-next-line reentrancy-events
+            IL2StandardERC721(_l2Token).mint(_to, _tokenId);
+            // slither-disable-next-line reentrancy-events
+            emit DepositFinalized(_l1Token, _l2Token, _from, _to, _tokenId, _data);
+        } else {
+            // Either the L2 token which is being deposited-into disagrees about the correct address
+            // of its L1 token, or does not support the correct interface.
+            // This should only happen if there is a  malicious L2 token, or if a user somehow
+            // specified the wrong L2 token address to deposit into.
+            // In either case, we stop the process here and construct a withdrawal
+            // message so that users can get their NFT out in some cases.
+            // There is no way to prevent malicious token contracts altogether, but this does limit
+            // user error and mitigate some forms of malicious contract behavior.
+            bytes memory message = abi.encodeWithSelector(
+                IL1ERC721Bridge.finalizeERC721Withdrawal.selector,
+                _l1Token,
+                _l2Token,
```

Comment:
It'd be easy to get the order of these addresses wrong wrong, esp because it kind of feels like they should be flipped along with `_to`, `_from`. FWIW I stared at them for a while, and they LGTM.